object-cache.php

```diff
@@ -1143,6 +1216,43 @@ protected function _call_redis( $method ) {
 
 	}
 
+	public function retry_exception_messages() {
+		$retry_exception_messages = array( 'socket error on read socket', 'Connection closed', 'Redis server went away' );
+		return apply_filters( 'wp_redis_retry_exception_messages', $retry_exception_messages );
+	}
+
+	public function message_matches( $error, $errors ) {
+		foreach ( $errors as $message ) {
+			$pattern = $this->_format_message_for_pattern( $message );
+			$matches = (bool) preg_match( $pattern, $error );
+			if ( $matches ) {
```

Comment:
I added this because Predis' exception messages include details about the connection, so instead of having to muck about trying to figure out how to get connection details in the messages, I figured it was easier and simpler to allow the error messages to actually be regex patterns we can test against